plugins/VersionUpgrade/VersionUpgrade33to34/VersionUpgrade33to34.py

```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2018 Ultimaker B.V.
+# Cura is released under the terms of the LGPLv3 or higher.
+
+import configparser #To parse preference files.
+import io #To serialise the preference files afterwards.
+
+from UM.VersionUpgrade import VersionUpgrade #We're inheriting from this.
+
+
+##  Upgrades configurations from the state they were in at version 3.2 to the
+#   state they should be in at version 3.3.
+class VersionUpgrade33to34(VersionUpgrade):
+
+    ##  Gets the version number from a CFG file in Uranium's 3.2 format.
+    #
+    #   Since the format may change, this is implemented for the 3.2 format only
+    #   and needs to be included in the version upgrade system rather than
+    #   globally in Uranium.
+    #
+    #   \param serialised The serialised form of a CFG file.
+    #   \return The version number stored in the CFG file.
+    #   \raises ValueError The format of the version number in the file is
+    #   incorrect.
+    #   \raises KeyError The format of the file is incorrect.
+    def getCfgVersion(self, serialised):
+        parser = configparser.ConfigParser(interpolation = None)
+        parser.read_string(serialised)
+        format_version = int(parser.get("general", "version")) #Explicitly give an exception when this fails. That means that the file format is not recognised.
+        setting_version = int(parser.get("metadata", "setting_version", fallback = 0))
+        return format_version * 1000000 + setting_version
+
+    ##  Upgrades instance containers to have the new version
+    #   number.
+    def upgradeInstanceContainer(self, serialized, filename):
+        parser = configparser.ConfigParser(interpolation = None)
+        parser.read_string(serialized)
+
+        # Update version number.
+        parser["general"]["version"] = "4"
```

Comment:
If all we do is updating the version number, why do we need to upgrade at all?

Reply:
No idea. Why did you add it @LipuFei?

Reply:
To trigger a version upgrade so we can move and rename the existing custom profiles, materials, etc. to have their own namespaces (dirs).